examples/nemo/eg5/Makefile

```diff
@@ -0,0 +1,103 @@
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# ------------------------------------------------------------------------------
+# Author: A. R. Porter, STFC Daresbury Lab
+# Modified J. Henrichs, Bureau of Meteorology
+
+# Makefile for the 5th NEMO example. Uses PSyclone (which must
+# be installed) to generate Fortran for the tracer-advection benchmark.
+# The script provided here instrument the kernels
+# with code to extract input- and output-data into a data file.
+
+# The compiler to use may be specified via the F90 environment variable,
+# it defaults to:
+# export F90=gfortran
+# export F90FLAGS="-g -O0"
+
+
+include ../../common.mk
+
+TYPE ?= standalone
+
+GENERATED_FILES += psy.f90 psy.o traadv-$(TYPE).exe output.dat psyclone.log \
+  traldf_iso.F90
+
+F90FLAGS ?= -fopenmp
+LDFLAGS ?= -fopenmp
+
+# Pick up the right extraction library:
+PSYROOT=../../..
+ifeq ($(TYPE), netcdf)
+	EXTRACT_DIR ?= $(PSYROOT)/lib/extract/netcdf/nemo
+	READ_DIR ?= $(PSYROOT)/lib/extract/netcdf
+	F90FLAGS += $$(nf-config --fflags)
+	LDFLAGS += $$(nf-config --flibs)
+	GENERATED_FILES += tra_adv-r?.nc
+else
+	EXTRACT_DIR ?= $(PSYROOT)/lib/extract/standalone/nemo
+	READ_DIR ?= $(PSYROOT)/lib/extract/standalone
+	GENERATED_FILES += tra_adv-r?.binary
+endif
+
+F90FLAGS += -I$(EXTRACT_DIR) -I$(READ_DIR)
+LIB_NAME = lib_extract.a
+
+
+.PHONY: allclean
+
+compile: traadv-$(TYPE).exe
+
+run: compile
+	IT=2 JPI=10 JPJ=10 JPK=5  ./traadv.exe
+
+traadv-$(TYPE).exe: psy.o $(EXTRACT_DIR)/$(LIB_NAME)
+	$(F90) psy.o -o traadv-$(TYPE).exe $(EXTRACT_DIR)/$(LIB_NAME) $(LDFLAGS)
+
+transform: kernels
+
+# Need `-l all` to ensure line-lengths in generated code are less than the
+# standard-mandated 132 chars. Also kernel extraction does not support
```

Comment:
Since this is NEMO you don't need the -nodm flag and associated comment.

Reply:
Well, I actually do need it ... but I totally agree that this is stupid. So I've fixed this, the distributed memory flag is now only tested if the API is not NEMO. I've also added tests to see it works as expected with LFRic/GOcean, and a completely new test for nemo. I didn't add any more nemo specific tests for nemo - I expect that Nemo will always work with the base class (since it's pure PSyIR). At this stage I can't decide if we really want a new `nemo_extract_trans.py` file that just adds this one test (the test for DM is actually important at this stage for the extraction libraries, they do not support MPI). We would then have tests for distributed memory duplicated in any API-specific implementation, which is ugly as well. I suggest we leave it till the extraction libraries support MPI (TODO #1992), then it will just go away.

Reply:
Oops.